vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/HttpFeedClient.java

```diff
@@ -113,31 +120,57 @@ private CompletableFuture<Result> send(String method, DocumentId documentId, Str
         return promise;
     }
 
+    private void verifyConnection(FeedClientBuilderImpl builder) {
+        if (builder.dryrun) return;
+        try (Cluster cluster = new ApacheCluster(builder)) {
+            HttpRequest request = new HttpRequest("POST",
+                                                  getPath(DocumentId.of("feeder", "handshake", "dummy")) + getQuery(empty(), true),
+                                                  requestHeaders,
+                                                  null,
+                                                  Duration.ofSeconds(10));
+            CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+            cluster.dispatch(request, future);
+            HttpResponse response = future.get(20, TimeUnit.SECONDS);
+            if (response.code() != 200) {
+                throw new FeedException("non-200 response: " + response);
```

Comment:
Create a proper feed exception by parsing the response and include status code with response message:
1) If `application/json` try to parse the message field
2) If `text/plain` use response content raw

Fallback to `toString()` otherwise.

Reply:
PTAL.